stdlib/javascript/compile.mc

```diff
@@ -121,56 +210,104 @@ lang MExprJSCompile = MExprAst + JSProgAst
-      -- Else compile each expression in sequence and return a list
-      let tms: [JSExpr] = map compileExpr tms in
-      JSESeq { exprs = tms, info = info }
-
-  | TmRecordUpdate _ -> error "Record updates cannot be handled in compileExpr."
-  | TmConApp _ -> error "Constructor application in compileExpr."
-  | TmLet _ -> error "Let expressions cannot be handled in compileExpr."
-  | TmRecLets _ -> error "Recursive let expressions cannot be handled in compileExpr."
-  | TmType _ -> error "Type expressions cannot be handled in compileExpr."
-  | TmConDef _ -> error "Constructor definitions cannot be handled in compileExpr."
-  | TmMatch _ -> error "Match expressions cannot be handled in compileExpr."
-  | TmUtest _ -> error "Unit test expressions cannot be handled in compileExpr."
-  | TmExt _ -> error "External expressions cannot be handled in compileExpr."
+      JSEArray { exprs = map (compileMExpr opts) tms }
 
   -- Literals
   | TmConst { val = val } ->
     match val      with CInt   { val = val } then JSEInt   { i = val }
     else match val with CFloat { val = val } then JSEFloat { f = val }
     else match val with CChar  { val = val } then JSEChar  { c = val }
     else match val with CBool  { val = val } then JSEBool  { b = val }
+    else match compileCOp opts [] val with jsexpr then jsexpr -- SeqOpAst Consts are handled by the compile operator semantics
     else error "Unsupported literal"
+  | TmRecordUpdate _ -> error "Record updates cannot be handled in compileMExpr."
+
+
+  ----------------
+  -- STATEMENTS --
+  ----------------
+
+  | TmLet { ident = id, body = expr, inexpr = e } ->
+    -- Check if identifier is the ignore identifier (_, or [])
+    -- If so, ignore the expression unless it is a function application
+    let data = (match nameGetStr id with [] then
+      match expr with TmApp { } then -- Inline the function call
+        ([(compileMExpr opts) expr], (compileMExpr opts) e)
+      else -- Ignore the expression
+        ([], (compileMExpr opts) e)
+    else -- Normal let binding
+      ([JSEDef { id = id, expr = (compileMExpr opts) expr }], (compileMExpr opts) e)
+    ) in
+    match data with (exprs, ret) then
+      flattenBlock (JSEBlock {
+        exprs = exprs,
+        ret = ret
+      })
+    else never
+
+  | TmRecLets { bindings = bindings, inexpr = e } ->
+    -- Todo: Compile each binding as a let expression
+    flattenBlock (JSEBlock {
+      exprs = map (
+        lam bind : RecLetBinding.
+        match bind with { ident = ident, body = body } then
+          let nop = TmConst { val = CFlushStdout { } } in
+          compileMExpr opts (TmLet { ident = ident, body = body, inexpr = nop })
+        else never
+        ) bindings,
+      ret = (compileMExpr opts) e
+    })
+  | TmType { inexpr = e } -> (compileMExpr opts) e -- no op (Skip type declaration)
+  | TmConApp _ -> error "Constructor application in compileMExpr."
```

Comment:
In general, it's better to use `infoErrorExit` and pass the `info` field of the expression, when an `info` is available. This shows where in the file you're compiling the error originates from, instead of where in the compiler the error occured (which is pointless to a user). If #590 is merged before this PR, I think you could update this line to something like (I don't remember what `err` will be called though)
```suggestion
  | TmConApp t -> errorSingle {{err with info = t.info} with msg = "Constructor application in compileMExpr."}
```
Otherwise, I suggest you use the `infoErrorExit` (which is being replaced in that PR, but it's still better than using `error`) as follows
```suggestion
  | TmConApp t -> infoErrorExit t.info "Constructor application in compileMExpr."
```

Reply:
Please do this for all places where you use `error` and it isn't too much work to get access to an `info` field (all `Expr` have an `info` field).

Reply:
Since #590 has been merged now, I suggest that you use the `errorSingle` function:
```suggestion
  | TmConApp _ -> errorSingle [t.info] "Constructor application in compileMExpr."
```